vm/src/thread.rs

```diff
@@ -560,7 +564,11 @@ impl ThreadInternal for Thread {
 
 
     /// Calls a module, allowed to to run IO expressions
-    fn call_module(&self, typ: &ArcType, closure: GcPtr<ClosureData>) -> Result<Value> {
+    fn call_module(&self,
+                   typ: &ArcType,
+                   closure: GcPtr<ClosureData>,
+                   run_io: bool)
+                   -> Result<Value> {
```

Comment:
It might be better to have two functions, rather than the boolean flag. `call_module` and `call_io_module`?